### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by IvarsKarpics**
* replaces HardwareRepository with mxcubecore
* added qt_import.py to utils directory. Importing from mxcubegui is strange and easiest solution is to duplicated the qt_import.py. This is open for discussion.

PR is compatible with latest changes in mxcube/mxcube

### CODE REVIEW

File: HardwareObjects/EMBL/EMBLXrayImaging.py [hunk 1]
```diff
@@ -38,14 +38,14 @@
 from cStringIO import StringIO
 from PIL.ImageQt import ImageQt
 
-from gui.utils import QtImport, Colors
+from gui.utils import qt_import, Colors
```
Comment (rhfogh) [inline HardwareObjects/EMBL/EMBLXrayImaging.py:41]: Change to mxcubecore.utils?

File: HardwareObjects/QtAxisCamera.py [hunk 2]
```diff
@@ -29,14 +29,9 @@
...
 
-try:
-    from PyQt5.QtGui import QPainter, QPixmap, QPen, QBrush, QImage
-    from PyQt5.QtCore import Qt
-except ImportError:
-    from PyQt4.QtGui import QPainter, QPixmap, QPen, QBrush, QImage
-    from PyQt4.QtCore import Qt
-
-from HardwareRepository.HardwareObjects.abstract.AbstractVideoDevice import (
+from mxcubeqt.utils import qt_import
```
Comment (rhfogh) [inline HardwareObjects/QtAxisCamera.py:32]: Change to mxcubecore.utils?

File: HardwareObjects/DESY/MjpgStreamVideo.py [hunk 3]
```diff
@@ -33,10 +33,10 @@
 import json
 
 # from PyQt4.QtGui import QImage, QPixmap
-from gui.utils.QtImport import QImage, QPixmap
-from HardwareRepository.ConvertUtils import string_types
+from gui.utils.qt_import import QImage, QPixmap
```
Comment (rhfogh) [inline HardwareObjects/DESY/MjpgStreamVideo.py:36]: Change to mxcubecore.utils?

**Comment by marcus-oscarsson** [comment]
Yes, I think that is the conclusion of one of our previous discussion. Thanks for paying attention to that,

File: HardwareObjects/SOLEIL/PX2/PX2Qt4_LimaVideo.py [hunk 4]
```diff
@@ -110,10 +110,10 @@ def set_exposure_time(self, exposure_time):
 
 
 def test_hwo():
-    from gui.utils.QtImport import *
+    from gui.utils.qt_import import *
```
Comment (rhfogh) [inline HardwareObjects/SOLEIL/PX2/PX2Qt4_LimaVideo.py:113]: Change to mxcubecore.utils?

Get rid of 'import *'?

File: HardwareObjects/SOLEIL/PX2/PX2Video.py [hunk 5]
```diff
@@ -50,7 +50,7 @@
 from GenericVideoDevice import GenericVideoDevice
 from camera import camera
 
-from gui.utils.QtImport import QImage, QPixmap
+from gui.utils.qt_import import QImage, QPixmap
```
Comment (rhfogh) [inline HardwareObjects/SOLEIL/PX2/PX2Video.py:53]: Change to mxcubecore.utils?

File: HardwareObjects/VaporyVideo.py [hunk 6]
```diff
@@ -23,9 +23,9 @@
 import time
 import gevent
 import vapory
-from gui.utils.QtImport import QImage
-from HardwareRepository import BaseHardwareObjects
-from HardwareRepository.HardwareObjects.Camera import JpegType
+from gui.utils.qt_import import QImage
```
Comment (rhfogh) [inline HardwareObjects/VaporyVideo.py:26]: Change to mxcubecore.utils?

File: HardwareObjects/VimbaVideo.py [hunk 7]
```diff
@@ -8,8 +8,8 @@
 except ImportError:
     pass
 
-from gui.utils.QtImport import QImage, QPixmap
-from HardwareRepository.HardwareObjects.abstract.AbstractVideoDevice import (
+from gui.utils.qt_import import QImage, QPixmap
```
Comment (rhfogh) [inline HardwareObjects/VimbaVideo.py:11]: Change to mxcubecore.utils?

File: deprecated/HardwareObjects/mockup/VideoMockup.py [hunk 8]
```diff
@@ -4,9 +4,9 @@
 import os
 import time
 import gevent
-from gui.utils.QtImport import QImage
-from HardwareRepository import BaseHardwareObjects
-from HardwareRepository.HardwareObjects.Camera import JpegType
+from gui.utils.qt_import import QImage
```
Comment (rhfogh) [inline deprecated/HardwareObjects/mockup/VideoMockup.py:7]: Change to mxcubecore.utils?

**Comment by marcus-oscarsson** [comment]
And thanks @IvarsKarpics for doing this, I thought we said that Rasmus and myself would take care of it but this is just as good.

**Review by rhfogh — CHANGES_REQUESTED** [review]
Approve subject to commented changes

**Comment by marcus-oscarsson** [comment]
I think we need to have a look at why the tests are failing

**Comment by IvarsKarpics** [comment]
Hi @marcus-oscarsson  and @rhfogh  I fixed qt_import and pytest.

File: HardwareObjects/VaporyVideo.py (hunk 6 above)
Comment (IvarsKarpics) [reply HardwareObjects/VaporyVideo.py]: done

**Comment by IvarsKarpics** [comment]
Hi @marcus-oscarsson  and @rhfogh 

Are you ok with the PR. Can we merge it and move further?

**Comment by rhfogh** [comment]
I am OK.
Yours,
Rasmus

**Comment by IvarsKarpics** [comment]
Excellent!!! Thank you!

**Pull request merged**